examples/hybrid/plane/topo_agnesi_nh.jl

```diff
@@ -0,0 +1,427 @@
+    )
+    # we want the total u³ at the boundary to be zero: we can either constrain
+    # both to be zero, or allow one to be non-zero and set the other to be its
+    # negation
+
+    # explicit part
+    dρ .-= vdivf2c.(Ic2f.(cρ .* cuₕ))
+    # implicit part
+    dρ .-= vdivf2c.(Ic2f.(cρ) .* fw)
+
+    # 2) Momentum equation
+
+    # curl term
+    hcurl = Operators.Curl()
+    # effectively a homogeneous Dirichlet condition on u₁ at the boundary
+    vcurlc2f = Operators.CurlC2F(
+        bottom = Operators.SetCurl(Geometry.Contravariant2Vector(0.0)),
+        top = Operators.SetCurl(Geometry.Contravariant2Vector(0.0)),
+    )
+
+    fω¹ = hcurl.(fw)
+    fω¹ .+= vcurlc2f.(cuₕ)
+
+    # cross product
+    # convert to contravariant
+    # these will need to be modified with topography
+    #    fu¹ =
+    #        Geometry.Contravariant1Vector.(Geometry.Covariant13Vector.(Ic2f.(cuₕ)),)
+    #    fu³ = Geometry.Contravariant3Vector.(Geometry.Covariant13Vector.(fw))
+    fu =
+        Geometry.Contravariant13Vector.(Ic2f.(cuₕ)) .+
+        Geometry.Contravariant13Vector.(fw)
+    fu¹ = Geometry.project.(Ref(Geometry.Contravariant1Axis()), fu)
+    fu³ = Geometry.project.(Ref(Geometry.Contravariant3Axis()), fu)
+    @. dw -= fω¹ × fu¹ # Covariant3Vector on faces
+    @. duₕ -= If2c(fω¹ × fu³)
+
+
+    @. duₕ -= hgrad(cp) / cρ
+    vgradc2f = Operators.GradientC2F(
+        bottom = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+        top = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+    )
+    @. dw -= vgradc2f(cp) / Ic2f(cρ)
+
+    cE = @. (norm(cuw)^2) / 2 + Φ(z)
+    @. duₕ -= hgrad(cE)
+    @. dw -= vgradc2f(cE)
+
+    # 3) total energy
+
+    @. dρe -= hdiv(cuw * (cρe + cp))
+    @. dρe -= vdivf2c(fw * Ic2f(cρe + cp))
+    @. dρe -= vdivf2c(Ic2f(cuₕ * (cρe + cp)))
+
+    # Uniform 2nd order diffusion
+    ∂c = Operators.GradientF2C()
+    fρ = @. Ic2f(cρ)
+    κ₂ = kinematic_viscosity # m^2/s
+
+    ᶠ∇ᵥuₕ = @. vgradc2f(cuₕ.components.data.:1)
+    ᶜ∇ᵥw = @. ∂c(fw.components.data.:1)
+    ᶠ∇ᵥh_tot = @. vgradc2f(h_tot)
+
+    ᶜ∇ₕuₕ = @. hgrad(cuₕ.components.data.:1)
+    ᶠ∇ₕw = @. hgrad(fw.components.data.:1)
+    ᶜ∇ₕh_tot = @. hgrad(h_tot)
+
+    hκ₂∇²uₕ = @. hwdiv(κ₂ * ᶜ∇ₕuₕ)
+    vκ₂∇²uₕ = @. vdivf2c(κ₂ * ᶠ∇ᵥuₕ)
+    hκ₂∇²w = @. hwdiv(κ₂ * ᶠ∇ₕw)
+    vκ₂∇²w = @. vdivc2f(κ₂ * ᶜ∇ᵥw)
+    hκ₂∇²h_tot = @. hwdiv(cρ * κ₂ * ᶜ∇ₕh_tot)
+    vκ₂∇²h_tot = @. vdivf2c(fρ * κ₂ * ᶠ∇ᵥh_tot)
+
+    dfw = dY.w.components.data.:1
+    dcu = dY.uₕ.components.data.:1
+
+    # Laplacian Diffusion (Uniform)
+    @. dcu += hκ₂∇²uₕ
+    @. dcu += vκ₂∇²uₕ
+    @. dfw += hκ₂∇²w
+    @. dfw += vκ₂∇²w
+    @. dρe += hκ₂∇²h_tot
+    @. dρe += vκ₂∇²h_tot
+
+    # Sponge tendency
+    β = @. rayleigh_sponge_z(coords.z)
+    βx = @. rayleigh_sponge_x(coords.x)
+    @. duₕ -= β * (uₕ - u_init)
+    @. dw -= Ic2f(β) * fw
+    #@. duₕ -= βx * (uₕ - u_init)
+    #@. dw -= Ic2f(βx) * fw
```

Comment:
I would remove stale comments and the related unused `rayleigh_sponge_x` function